app/models/service_ansible_playbook.rb

```diff
@@ -14,6 +14,25 @@ def preprocess(action, add_options = {})
   end
 
   def execute(action)
+    task_opts = {
+      :action => "Executing Ansible Playbook",
+      :userid => "system"
+    }
+
+    queue_opts = {
+      :args        => [action],
+      :class_name  => self.class.name,
+      :instance_id => id,
+      :method_name => "_execute",
+      :role        => "embedded_ansible"
+    }
+
+    task_id = MiqTask.generic_action_with_callback(task_opts, queue_opts)
+    task = MiqTask.wait_for_taskid(task_id)
```

Comment:
I don't think this should wait for task here, because that is effectively a blocking call (sleep loop, if I recall).  I expected a new state that uses queue based waiting

Reply:
The previous implementation didn't have a queue call at all so the blocking implementation is the same as it was previously.

Reply:
Maybe put another way, there's a check_complete method, and we should probably check for the task completion there, else bail for the next check

Reply:
At this stage the job itself was queued wasnt it?  That was my intention at least, and then the "externally running" job was supposed to be checked in check_complete

Reply:
> At this stage the job itself was queued wasnt it?

That's what I was trying to figure out, but got lost in automate. Here's `puts caller.join("\n")` at the start of the old `#execute` method:
```
/home/ncarboni/.gem/ruby/2.5.1/bundler/gems/manageiq-automation_engine-09c8bbdf6762/lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_model_base.rb:333:in `public_send'
/home/ncarboni/.gem/ruby/2.5.1/bundler/gems/manageiq-automation_engine-09c8bbdf6762/lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_model_base.rb:333:in `block in object_send'
/home/ncarboni/.gem/ruby/2.5.1/bundler/gems/manageiq-automation_engine-09c8bbdf6762/lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_model_base.rb:352:in `ar_method'
/home/ncarboni/.gem/ruby/2.5.1/bundler/gems/manageiq-automation_engine-09c8bbdf6762/lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_model_base.rb:362:in `ar_method'
/home/ncarboni/.gem/ruby/2.5.1/bundler/gems/manageiq-automation_engine-09c8bbdf6762/lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_model_base.rb:331:in `object_send'
/home/ncarboni/.gem/ruby/2.5.1/bundler/gems/manageiq-automation_engine-09c8bbdf6762/lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_model_base.rb:182:in `block (3 levels) in expose'
/home/ncarboni/Source/manageiq/app/models/user.rb:283:in `with_user'
/home/ncarboni/.gem/ruby/2.5.1/bundler/gems/manageiq-automation_engine-09c8bbdf6762/lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_model_base.rb:182:in `block (2 levels) in expose'
/home/ncarboni/.rubies/ruby-2.5.1/lib/ruby/2.5.0/drb/drb.rb:1630:in `perform_without_block'
/home/ncarboni/.rubies/ruby-2.5.1/lib/ruby/2.5.0/drb/drb.rb:1590:in `perform'
/home/ncarboni/.rubies/ruby-2.5.1/lib/ruby/2.5.0/drb/drb.rb:1674:in `block (2 levels) in main_loop'
/home/ncarboni/.rubies/ruby-2.5.1/lib/ruby/2.5.0/drb/drb.rb:1670:in `loop'
/home/ncarboni/.rubies/ruby-2.5.1/lib/ruby/2.5.0/drb/drb.rb:1670:in `block in main_loop'
```

So up to this point the service was queued for execution, but for the automate role. I now need the job itself to run on the appliance with the embedded_ansible role.

[`ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Job.create_job`](https://github.com/ManageIQ/manageiq/blob/82fe63826beb4fa43a472fc93d7ee3fd9e800e6e/app/models/service_ansible_playbook.rb#L28) calls [run on the passed job template](https://github.com/ManageIQ/manageiq/blob/82fe63826beb4fa43a472fc93d7ee3fd9e800e6e/app/models/manageiq/providers/embedded_ansible/automation_manager/job.rb#L28) which in turn creates the workflow and [calls signal the first time](https://github.com/ManageIQ/manageiq/blob/82fe63826beb4fa43a472fc93d7ee3fd9e800e6e/app/models/manageiq/providers/embedded_ansible/automation_manager/configuration_script.rb#L45-L47), but the base signal implementation just calls the method in the same process and `AnsibleRunnerWorkflow#queue_signal` is set up to run the method on the same server [here](https://github.com/ManageIQ/manageiq/blob/82fe63826beb4fa43a472fc93d7ee3fd9e800e6e/app/models/manageiq/providers/ansible_runner_workflow.rb#L62-L78).

So to make a long story short, I'm pretty sure this is what we need to do :sweat_smile:

Reply:
I think my expectation of what create_job did was wrong, but I still contend that this should be async.  In good with merge for now though and talking about that for follow-up since it's technically the same before/after as you've said